### DISCUSSION TITLE
Revision, audit and securing of permission changes.

### DISCUSSION THREAD

**Discussion by u12206050** [directus/directus#12966] [Feature Requests]
Due to the low-code architecture of Directus, roles and permissions become effectively the most important business logic decisions to make when setting up your api.

Shockingly whilst everything else has support for revisions and an activity log, permissions do not.

So here is a short list in order of what could be invaluable in the long run.

- [ ] Revision history of changes per permission per role
- [ ] Refactor to not just save permissions once edited, but need to "save role" or "Publish" in order for new permissions to take affect.
- [ ] *Run automated api query with affected role and permissions to display to admin what affect the permission has, can greatly help with making it clear what still can and can not be done.

*The last one could be something in of itself, so feel free to ignore if it doesn't sound to cool :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
